src/object/cli_obj.c

```diff
@@ -5865,6 +5865,8 @@ daos_obj_generate_oid(daos_handle_t coh, daos_obj_id_t *oid,
 {
 	daos_ofeat_t	feat = (daos_ofeat_t)in;
 
+	D_WARN("daos_ofeat_t(DAOS_OF_XXX) will be deprecated soon, "
+		"please use enum daos_otype_t(DAOS_OT_XXX) instead!\n");
```

Comment:
Shall we mark this function as __attribute__((deprecated)) to generate a warning at compile time? Maybe that would be super clear actually ....

Reply:
This might not work for current DAOS, as it will make compile failed:

src/object/cli_obj.c:5881:2: error: ‘daos_obj_generate_oid’ is deprecated (declared at src/object/cli_obj.c:5862) [-Werror=deprecated-declarations]
  return daos_obj_generate_oid(coh, oid, (enum daos_otype_t)feat, cid,
  ^
cc1: all warnings being treated as errors

Reply:
Hmm, a further think about this, i think warning message here might be too much. we might need a helper function like
D_WARN_ONCE()， D_WARN_RATED() to limit messages, otherwise such kind of messages could pollute logs files easily.